flang/unittests/Optimizer/Builder/ComplexTest.cpp

```diff
@@ -0,0 +1,75 @@
+//===- ComplexExprTest.cpp -- ComplexExpr unit tests ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Builder/Complex.h"
+#include "gtest/gtest.h"
+#include "flang/Optimizer/Builder/FIRBuilder.h"
+#include "flang/Optimizer/Support/InitFIR.h"
+#include "flang/Optimizer/Support/KindMapping.h"
+
+struct ComplexTest : public testing::Test {
+public:
+  void SetUp() override {
+    fir::KindMapping kindMap(&context);
+    mlir::OpBuilder builder(&context);
+    auto loc = builder.getUnknownLoc();
+
+    // Set up a Module with a dummy function operation inside.
+    // Set the insertion point in the function entry block.
+    mlir::ModuleOp mod = builder.create<mlir::ModuleOp>(loc);
+    mlir::FuncOp func = mlir::FuncOp::create(
+        loc, "func1", builder.getFunctionType(llvm::None, llvm::None));
+    auto *entryBlock = func.addEntryBlock();
+    mod.push_back(mod);
+    builder.setInsertionPointToStart(entryBlock);
+
+    fir::support::loadDialects(context);
+    firBuilder = std::make_unique<fir::FirOpBuilder>(mod, kindMap);
+  }
+
+  fir::FirOpBuilder &getBuilder() { return *firBuilder; }
```

Comment:
[nit] This is not needed, `firBuilder` is a public member.

Also, why not simplify this and make `firBuilder` a regular member rather than a pointer? `SetUp` is run for every test anyway.

Reply:
`getBuilder` is a common pattern used in all other tests in the builder directory. If I remove it from here and directly use the `firbuilder`, it will be inconsistent with the others.
I think we can have a follow-up patch to update all of them.
I think `firBuilder` is a pointer so that it can be initialized in `Setup`. If we make it otherwise I think it will be default constructed and then we will have to make another construction in `Setup`.